main.js

```diff
@@ -2,95 +2,133 @@
 
 const todos = [];
 const ul = document.querySelector('ul');
+const table = document.querySelector('table');
 let statusButton,deleteButton;
 
 
-function addTodo (word) {
+function addTodo(word) {
   const addItem = {};
   addItem[word] = 'working';
   todos.push(addItem) ;
 }
 
+function createTd(todo, index) {
+  const tr = document.createElement('tr');
+  const td1 = document.createElement('td');
+  td1.innerHTML = index;
+  tr.appendChild(td1);
+  
+  const td2 = document.createElement('td');
+  td2.innerHTML = Object.keys(todo);
+  tr.appendChild(td2);
+  
+  createDeleteButton();
+  addDeleteFeature(index);
+  tr.appendChild(deleteButton);
+  
+  createStatusButton(Object.values(todo));
+  addStatusFeature(Object.keys(todo), index);
+  tr.appendChild(statusButton);
+
+  table.appendChild(tr);
+}
+
+function showFinish() {
+  updateStatus();
+  todos.forEach((todo, index) => {
+    if (`${todos[index][Object.keys(todo)]}`.includes('finish')) {
+      createTd(todo, index);
+    }
+  });
+}
 
-function showTodos () {
+function showWorking() {
+  updateStatus();
   todos.forEach((todo, index) => {
-    const li = document.createElement('li');
-    li.innerHTML = index;
-    
-    const span = document.createElement('span');
-    span.innerHTML = Object.keys(todo);
-    li.appendChild(span);
-    
-    createDeleteButton ();
-    addDeleteFeature (index);
-    li.appendChild(deleteButton);
-    
-    createStatusButton (Object.values(todo));
-    addStatusFeature (Object.keys(todo), index);
-    li.appendChild(statusButton);
-    
-    ul.appendChild(li);
-   });
+    if (todos[index][Object.keys(todo)] === 'working') {
+      createTd(todo, index);
+    }
+  });
 }
 
+function showTodos() {
+  updateStatus();
+  todos.forEach((todo, index) => {
+    createTd(todo, index);
+ });
+}
 
-function createStatusButton (status) {
+function createStatusButton(status) {
   statusButton = document.createElement('button');
-  if (status == 'working finish') {
-    statusButton.className = 'working finish';
+  if(status == 'working finish') {
```

Comment:
やや細かいですが、比較は厳密にしておきましょう。

```suggestion
  if(status === 'working finish') {
```